File: app/src/main/java/com/example/tipcalculator/BillFormatter.java
package com.example.tipcalculator;


import java.util.StringTokenizer;

public class BillFormatter {

    public final static String CURRENCY_SYMBOL = "$";

    private Bill bill;

    public BillFormatter(String str_amount, int tipPercent, int nPerson) throws Bill.ZeroAmountException {
        Double amount = getAmount(str_amount);

        bill = new Bill(amount, tipPercent, nPerson);
    }

    public static String getPersonString(int nP) {
        String out = "People:"+nP;
        return out;
    }

    public static String getTipString(int tip) {
        String out = "Tip "+tip+"%";
        return out;
    }

    private Double getAmount(String str_amount) {
        str_amount = str_amount.trim();
        double amount = Double.parseDouble(str_amount);
        return amount;
    }

    public String getTipAmount() {
        double tip = bill.getTipAmount();
        tip = BillFormatter.roundOffToTwoDigits(tip);

        String tipStr =  getStringFromMonetaryValue(tip);

        return tipStr;
    }

    public String getTotalAmount() {
        double total = bill.getTotalAmount();
        total = roundOffToTwoDigits(total);

        String totalStr =  getStringFromMonetaryValue(total);

        return totalStr;
    }

    public String getAmountPerPerson() {
        double amtPP = bill.getAmountPerPerson();
        amtPP = roundOffToTwoDigits(amtPP);

        String amtPP_Str =  getStringFromMonetaryValue(amtPP);

        return amtPP_Str ;
    }

    public static String getStringFromMonetaryValue(double amount){
        String amountStr = String.valueOf(amount);

        amountStr = CURRENCY_SYMBOL + amountStr;
        return amountStr;
    }


    public static double roundOffToTwoDigits(double d) {
        d *= 100;
        d = Math.round(d);
        d = d / 100;
        return d;
    }
}

File: app/src/main/java/com/example/tipcalculator/BillFormatter.kt
package com.example.tipcalculator

class BillFormatter(str_amount: String, tipPercent: Int, nPerson: Int) {
    private val bill: Bill
    private fun getAmount(str_amount: String): Double {
        var str_amount = str_amount
        str_amount = str_amount.trim { it <= ' ' }
        return str_amount.toDouble()
    }

    val tipAmount: String
        get() {
            var tip = bill.tipAmount
            tip = roundOffToTwoDigits(tip)
            return getStringFromMonetaryValue(tip)
        }

    val totalAmount: String
        get() {
            var total = bill.totalAmount
            total = roundOffToTwoDigits(total)
            return getStringFromMonetaryValue(total)
        }

    val amountPerPerson: String
        get() {
            var amtPP = bill.amountPerPerson
            amtPP = roundOffToTwoDigits(amtPP)
            return getStringFromMonetaryValue(amtPP)
        }

    companion object {
        const val CURRENCY_SYMBOL = "$"
        @JvmStatic
        fun getPersonString(nP: Int): String {
            return "People:$nP"
        }

        @JvmStatic
        fun getTipString(tip: Int): String {
            return "Tip $tip%"
        }

        fun getStringFromMonetaryValue(amount: Double): String {
            var amountStr = amount.toString()
            amountStr = CURRENCY_SYMBOL + amountStr
            return amountStr
        }

        fun roundOffToTwoDigits(d: Double): Double {
            var d = d
            d *= 100.0
            d = Math.round(d).toDouble()
            d = d / 100
            return d
        }
    }

    init {
        val amount = getAmount(str_amount)
        bill = Bill(amount, tipPercent, nPerson)
    }
}